src/web/js/dashboard/GoogleAPI.js

```diff
@@ -93,31 +110,449 @@ class GoogleAPI {
   }
 
   saveAppData = (fileId, appData) => {
-    return window.gapi.client.drive.files.update({
+    return window.gapi.client.request({
       path: '/upload/drive/v3/files/' + fileId,
       method: 'PATCH',
       params: {
         uploadType: 'media'
       },
-      body: JSON.stringify(appData)
+      body: appData
     });
   }
 
   // Create and render a Google Picker object for selecting a file.
   createPicker = (callback) => {
     window.gapi.load('picker', function(){
-      window.picker = new window.google.picker.PickerBuilder()
+      var picker = new window.google.picker.PickerBuilder()
         .enableFeature(window.google.picker.Feature.MULTISELECT_ENABLED)
         .setTitle("Select a Pyret document")
-        .addView(new window.google.picker.View(window.google.picker.ViewId.DOCS))
+        .addView(new window.google.picker.View(window.google.picker.ViewId.DOCS).setQuery("*.arr"))
         .setOAuthToken(window.gapi.auth.getToken().access_token)
         .setCallback(callback)
-        .setOrigin(window.location.protocol + '//' + window.location.host)
+        .setOrigin(location.protocol + '//' + location.host)
         .build();
 
-      window.picker.setVisible(true);
+      picker.setVisible(true);
+    });
+  }
+
+  //keeping this around to keep calls clean in getters/setters
+  getPyretData = () => {
+    return this.getAppDataFileID().then((response) => {
+      var fileID = response.result.files[0].id;
+      return this.getAppDataFileContent(fileID);
     });
   }
-}
 
+  savePyretData = (newData) => {
+    return this.getAppDataFileID().then((response) => {
+      var fileID = response.result.files[0].id;
+      return this.saveAppData(fileID, newData);
+    });
+  }
+
+  /**
+  Intended to be used on login. If a pyret data file does not exist, it is created.
+  
+  JSON format for saving pyret data:
+  
+  pyretinfo.json{
+    //unique IDs for classes, students, and assignments.
+    //these are just incremented each time we make an assignment
+    nextClassID: int,
```

Comment:
Nope, if the JSON file gets reset, the nextClassID param will get repopulated to 0. This is just something that we use internally to access data in the JSON file, it has no relation to Google Drive folder IDs.